CoreSchemaView/src/au/gov/asd/tac/constellation/views/schemaview/providers/TransactionTypeNodeProvider.java

```diff
@@ -130,15 +143,98 @@ public void newActiveGraph(final Graph graph) {
             } else {
                 schemaLabel.setText("No schema available");
             }
+            populateTree();
+        });
+    }
 
-            final TreeItem<SchemaTransactionType> root = createNode(null);
-            treeView.setRoot(root);
+    private VBox addFilter() {
+        filterText.setPromptText("Filter transaction types");
+        final ToggleGroup tg = new ToggleGroup();
+        startsWithRb.setToggleGroup(tg);
+        startsWithRb.setPadding(new Insets(0, 0, 0, 5));
+        startsWithRb.setSelected(true);
+        final RadioButton containsRb = new RadioButton("Contains");
+        containsRb.setToggleGroup(tg);
+        containsRb.setPadding(new Insets(0, 0, 0, 5));
+
+        tg.selectedToggleProperty().addListener((ov, oldValue, newValue) -> {
+            populateTree();
         });
+
+        filterText.textProperty().addListener((ov, oldValue, newValue) -> {
+            populateTree();
+        });
+
+        final HBox headerBox = new HBox(new Label("Filter: "), filterText, startsWithRb, containsRb);
+        headerBox.setAlignment(Pos.CENTER_LEFT);
+        headerBox.setPadding(new Insets(5));
+
+        final VBox box = new VBox(schemaLabel, headerBox, treeView);
+        VBox.setVgrow(treeView, Priority.ALWAYS);
+        return box;
+    }
+
+    private void populateTree() {
+        final TreeItem<SchemaTransactionType> root = createNode(null);
+        treeView.setRoot(root);
+    }
+
+    private boolean isFilterMatchCurrentNode(SchemaTransactionType treeItem) {
+        boolean found = isFilterMatchText(treeItem.getName());
+        if (!found) {
+            found = isFilterMatchAnyProperty(treeItem);
+        }
+        return found;
```

Comment:
This can all be simplified to `return isFilterMatchText(treeItem.getName()) || isFilterMatchAnyProperty(treeItem)`

Reply:
I just tried to loop the properties only if it doesn't match the type name, to avoid unnecessary processing. May be worthless the effort in this scenario?

Reply:
@Auriga2 so what you had is fine though it could be simplified with @antares1470's suggestion as a 1 line return statement. Are you able to consider making this change?